### ISSUE TITLE
Plans to upgrade R.Net to R4.0

### ISSUE THREAD

**Issue by UnclAlDeveloper** [rdotnet/rdotnet#123]
What are the current plans to look into upgrading R.Net to be compatible with R4.0, and any sort of timeline?. I understand it would be a bit early to know, but just any sort of basic guidelines of what to expect?  I've asked the same question on the rClr repository also.

Thanks

**Comment by lrasmus** [comment]
Thanks @UnclAlDeveloper - I hadn't been tracking the release.  I'll read the release notes, do some testing with R 4.0, and see how things work.  We definitely want to maintain support for newer R versions.

**Comment by lrasmus** [comment]
I ran the unit tests to check how R.NET works with R 4.0.  I have [one assertion failing](https://github.com/rdotnet/rdotnet/blob/1.8.2/RDotNet.Tests/ListsTest.cs#L79), but I believe the failure is correct given a change in R 4.0, that the test just needs to be updated, and that users can proceed with R.NET and R 4.0.

Per the comments in the failing test, this is the intended R behavior that the test should reflect:
```
 > str(as.list(x))
   List of 2
    $ a: Factor w/ 3 levels "A","B","C": 1 2 3 1 2 3
    $ b: int [1:6] 1 2 3 1 2 3
```

We see that in R 3.6.3:
```
R version 3.6.3 (2020-02-29) -- "Holding the Windsock"

> x <- as.list(data.frame(a = rep(LETTERS[1:3], 2), b = rep(1:3, 2)))
> str(as.list(x))
List of 2
 $ a: Factor w/ 3 levels "A","B","C": 1 2 3 1 2 3
 $ b: int [1:6] 1 2 3 1 2 3
```

The output from R 4.0 is different, however:

```
R version 4.0.0 (2020-04-24) -- "Arbor Day"

> x <- as.list(data.frame(a = rep(LETTERS[1:3], 2), b = rep(1:3, 2)))
> str(as.list(x))
List of 2
 $ a: chr [1:6] "A" "B" "C" "A" ...
 $ b: int [1:6] 1 2 3 1 2 3
```

This makes sense given the following [release note for R 4.0.0](https://cran.r-project.org/doc/manuals/r-devel/NEWS.html):

> R now uses a stringsAsFactors = FALSE default, and hence by default no longer converts strings to factors in calls to data.frame() and read.table().

I've updated the tests to confirm that by default it's not converting strings to factors, and added a new set of assertions that it will convert to factor when you specify `stringsAsFactors = TRUE`.  I just put in PR #124 , but I don't think you need to wait for it to be merged.

**Comment by jmp75** [comment]
I've been unwell lately and this delayed a few things; [R.NET 1.9.0](https://www.nuget.org/packages/R.NET/1.9.0) is released and works on R 4.0.x. Also works around issue #127.
